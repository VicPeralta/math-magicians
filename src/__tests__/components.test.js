import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Key from '../components/key';
import KeyRow from '../components/keyRow';
import Calculator from '../components/calculator';
import ResultWindow from '../components/resultWindow';
import NavBar from '../components/navBar';

describe('Components testing', () => {
  test('Key Component', () => {
    const tree = renderer.create(<Key name="3" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('KeyRow Component', () => {
    const tree = renderer.create(<KeyRow keys={['AC', '+/-', '%', 'รท']} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Calculator Component', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ResultWindow Component', () => {
    const tree = renderer.create(<ResultWindow value="0" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('NavBar Component', () => {
    const Element = (
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>
    );
    const tree = renderer.create(Element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
