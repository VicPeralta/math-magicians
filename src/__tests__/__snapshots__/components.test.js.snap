// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components testing Calculator Component 1`] = `
<div
  className="calculator"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex="0"
>
  <div>
    <p
      className="results"
    >
      0
    </p>
  </div>
  <div
    className="keyRow"
  >
    <button
      className="key black "
      data-id="AC"
      type="button"
    >
      AC
    </button>
    <button
      className="key black "
      data-id="+/-"
      type="button"
    >
      +/-
    </button>
    <button
      className="key black "
      data-id="%"
      type="button"
    >
      %
    </button>
    <button
      className="key orange "
      data-id="รท"
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="keyRow"
  >
    <button
      className="key black "
      data-id="7"
      type="button"
    >
      7
    </button>
    <button
      className="key black "
      data-id="8"
      type="button"
    >
      8
    </button>
    <button
      className="key black "
      data-id="9"
      type="button"
    >
      9
    </button>
    <button
      className="key orange "
      data-id="x"
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="keyRow"
  >
    <button
      className="key black "
      data-id="4"
      type="button"
    >
      4
    </button>
    <button
      className="key black "
      data-id="5"
      type="button"
    >
      5
    </button>
    <button
      className="key black "
      data-id="6"
      type="button"
    >
      6
    </button>
    <button
      className="key orange "
      data-id="-"
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="keyRow"
  >
    <button
      className="key black "
      data-id="1"
      type="button"
    >
      1
    </button>
    <button
      className="key black "
      data-id="2"
      type="button"
    >
      2
    </button>
    <button
      className="key black "
      data-id="3"
      type="button"
    >
      3
    </button>
    <button
      className="key orange "
      data-id="+"
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="keyRow"
  >
    <button
      className="key black double"
      data-id="0"
      type="button"
    >
      0
    </button>
    <button
      className="key black "
      data-id="."
      type="button"
    >
      .
    </button>
    <button
      className="key orange "
      data-id="="
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Components testing Key Component 1`] = `
<button
  className="key black "
  data-id="3"
  type="button"
>
  3
</button>
`;

exports[`Components testing KeyRow Component 1`] = `
<div
  className="keyRow"
>
  <button
    className="key black "
    data-id="AC"
    type="button"
  >
    AC
  </button>
  <button
    className="key black "
    data-id="+/-"
    type="button"
  >
    +/-
  </button>
  <button
    className="key black "
    data-id="%"
    type="button"
  >
    %
  </button>
  <button
    className="key orange "
    data-id="รท"
    type="button"
  >
    รท
  </button>
</div>
`;

exports[`Components testing NavBar Component 1`] = `
<nav
  className="NavBar"
>
  <img
    alt="Math"
    src="maths.png"
  />
  <h1>
    Math Magicians
  </h1>
  <div>
    <a
      aria-current="page"
      className="link active"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "isActive": "active-link",
        }
      }
    >
      Home
    </a>
    <a
      className="link"
      href="/calculator"
      onClick={[Function]}
      style={
        Object {
          "isActive": "active-link",
        }
      }
    >
      Calculator
    </a>
    <a
      className="link"
      href="/quote"
      onClick={[Function]}
      style={
        Object {
          "isActive": "active-link",
        }
      }
    >
      Quote
    </a>
  </div>
</nav>
`;

exports[`Components testing ResultWindow Component 1`] = `
<div>
  <p
    className="results"
  >
    0
  </p>
</div>
`;
